src/components/EmojiReactions/EmojiReactions.tsx

```diff
@@ -0,0 +1,286 @@
+import { useChannel } from '@ably-labs/react-hooks'
+import { useEffect, useState } from 'react'
+import { useOutletContext } from 'react-router-dom'
+import defaultMessages, { EmojiUsage, Message } from './utils/messageData'
+import { RefreshIcon, EmojiHappyIcon } from '@heroicons/react/solid'
+import { Types } from 'ably'
+import type { ProjectInfo } from '../../Layout'
+
+const EmojiReactions = () => {
+  let { channelName, clientId, setProjectInfo } = useOutletContext<{
+    channelName: string
+    clientId: string
+    setProjectInfo: (projectInfo: ProjectInfo) => void
+  }>()
+
+  channelName = `reactions:${channelName}`
+  const emojis = ['😀', '❤️', '👋', '😹', '😡', '👏']
+  let usedEmojiCollection: EmojiUsage[] = []
+
+  const ADD_REACTION_EVENT = 'add-reaction'
+  const REMOVE_REACTION_EVENT = 'remove-reaction'
+  const SEND_EVENT = 'send'
+
+  const [addEmoji, setAddEmoji] = useState(true)
+
+  const [chatMessage, setChatMessage] = useState<Message>({})
+  const [showEmojiList, setShowEmojiList] = useState(false)
+
+  const [channel, ably] = useChannel(channelName, (msg) => {
+    switch (msg.name) {
+      case SEND_EVENT:
+        // reset reactions when new message is received
+        usedEmojiCollection = []
+        setChatMessage({
+          author: msg.data.author,
+          content: msg.data.content,
+          timeserial: msg.id,
+        })
+        break
+      case REMOVE_REACTION_EVENT:
```

Comment:
why do we handle this here rather than in getMessageReactions?

Reply:
getMessageReactions is only meant to subscribe to the reaction add event using the name and timeserial filter. This is following the message interaction documentation we have. Alternatively, it can be added to the main channel subscription but that sorts of take the user away from the message interaction feature we have